### PR TITLE
Add statements grammar component

### PR DESCRIPTION
**Pull request by matheussll**

### CODE REVIEW

File: Sources/HarmonizeSemantics/GrammarComponents/Provider/StatementsProviding.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//
+//  StatementsProviding.swift
+//  Harmonize
+//
+//  Created by Matheus Luz on 10/21/24.
+//
```
Comment (steliosfran) [inline Sources/HarmonizeSemantics/GrammarComponents/Provider/StatementsProviding.swift:6]: We probably need to add PSS here

File: Sources/HarmonizeSemantics/Declaration/Initializer.swift [hunk 2]
```diff
@@ -47,7 +48,11 @@ extension Initializer: AttributesProviding,
     public var body: String? {
         node.body?.statements.toString()
     }
-    
+
+    public var statements: [String] {
+        node.body?.statements.map { $0.trimmedDescription } ?? []
+    }
```
Comment (steliosfran) [inline Sources/HarmonizeSemantics/Declaration/Initializer.swift:54]: @lcszc do we need a test here?

Comment (lcszc) [reply Sources/HarmonizeSemantics/Declaration/Initializer.swift:54]: Yes.

But what do you think about creating a model for Body? As we now have statements (and we may have more properties in the future), we could put this into `BodyProviding`.

Even better, if we create a `Body` as a grammar component declaration we can do:

```swift
struct Body:  DeclarationDecoration {
  public let node: CodeBlockSyntax
  
  // ...
  
  var text: String
  
  var lines: String // or statements
}
```

What do you think? If that's too much I'd just recommend to move `statements` to `BodyProviding` and also to add your `statements` implementation for functions as well.

File: Sources/HarmonizeSemantics/GrammarComponents/Provider/StatementsProviding.swift (hunk 1 above)
Comment (lcszc) [reply Sources/HarmonizeSemantics/GrammarComponents/Provider/StatementsProviding.swift:6]: Yeah, to keep a pattern we can replace `Created by` with PSS copyright

**Review by lcszc — APPROVED** [review]

**Pull request merged**